### PR TITLE
Delete the various servers.

### PR DESCRIPTION
**Pull request by pphaneuf**
They have been unsupported for a while, so deleting them is the safest course of action, so that people are not tempted to use them.

### CODE REVIEW

**Comment by Martin2112** [comment]
Can we put a big pointer to the new stuff somewhere visible?

**Comment by pphaneuf** [comment]
There's already some links in the README: https://github.com/google/certificate-transparency#c-log-server-deprecation-notice

**Comment by Martin2112** [comment]
True. Maybe that's enough.

**Comment by pphaneuf** [comment]
Did update the README a bit, though.

File: configure.ac [hunk 1]
```diff
@@ -70,9 +70,6 @@ AC_CHECK_HEADER([google/protobuf/message.h],,
                 [AC_MSG_ERROR([protobuf headers could not be found])])
 AC_CHECK_HEADER([leveldb/db.h],,
                 [AC_MSG_ERROR([leveldb headers could not be found])])
-AC_CHECK_HEADER([sqlite3.h],,
-                [AC_MSG_ERROR([sqlite3 headers could not be found])])
-AC_CHECK_HEADER([ldns/ldns.h],, [missing_ldns=yes])
 AC_CHECK_HEADER([objecthash.h],, [missing_objecthash=yes])
```
Comment (daviddrysdale) [inline configure.ac]: Can objecthash be dinged too?

File: configure.ac [hunk 2]
```diff
@@ -70,9 +70,6 @@ AC_CHECK_HEADER([google/protobuf/message.h],,
                 [AC_MSG_ERROR([protobuf headers could not be found])])
 AC_CHECK_HEADER([leveldb/db.h],,
                 [AC_MSG_ERROR([leveldb headers could not be found])])
```
Comment (daviddrysdale) [inline configure.ac]: Don't think leveldb is used now, so can go from here and gclient config?

File: README.md [hunk 3]
```diff
@@ -96,7 +86,6 @@ The key subdirectories are:
  - For the main distributed CT Log itself:
    - `cpp/log`: Main distributed CT Log implementation.
    - `cpp/merkletree`: Merkle tree implementation.
-   - `cpp/server`: Top-level code for server implementations.
    - `cpp/monitoring`: Code to export operation statistics from CT Log.
  - The [CT mirror Log](docs/MirrorLog.md) implementation also uses:
    - `cpp/fetcher`: Code to fetch entries from another Log
```
Comment (daviddrysdale) [inline README.md]: `cpp/fetcher/` looks to get removed here; should it be? (I.e. does anything else use it?)  If so, drop this line too.

File: README.md [hunk 4]
```diff
@@ -96,7 +86,6 @@ The key subdirectories are:
  - For the main distributed CT Log itself:
    - `cpp/log`: Main distributed CT Log implementation.
    - `cpp/merkletree`: Merkle tree implementation.
-   - `cpp/server`: Top-level code for server implementations.
```
Comment (daviddrysdale) [inline README.md]: Still a couple of references to this:
```bash
% grep cpp/server test/*
test/run_mirror_server.sh:../cpp/server/ct-mirror \
test/sslconnect_test.sh:  ../cpp/server/ct-server --port=$log_server_port \
grep: test/testdata: Is a directory

```

File: configure.ac [hunk 5]
```diff
@@ -127,14 +124,6 @@ AS_IF([test -n "$missing_leveldb"],
       [AC_MSG_ERROR([could not find the leveldb/snappy libraries])])
```
Comment (daviddrysdale) [inline configure.ac]: Can we lose libsnappy too?

File: configure.ac (hunk 1 above)
Comment (pphaneuf) [reply configure.ac]: Yes, took it out now.

File: configure.ac (hunk 2 above)
Comment (pphaneuf) [reply configure.ac]: Gone!

File: README.md (hunk 4 above)
Comment (pphaneuf) [reply README.md]: Deleted!

File: README.md (hunk 3 above)
Comment (pphaneuf) [reply README.md]: Removed!

File: configure.ac (hunk 5 above)
Comment (pphaneuf) [reply configure.ac]: Dinged!

**Pull request merged**